lib/rubocop/cop/lint/void.rb

```diff
@@ -245,15 +245,23 @@ def autocorrect_nonmutating_send(corrector, node, suggestion)
         def entirely_literal?(node)
           case node.type
           when :array
-            node.each_value.all? { |value| entirely_literal?(value) }
+            all_values_entirely_literal?(node)
           when :hash
-            return false unless node.each_key.all? { |key| entirely_literal?(key) }
-
-            node.each_value.all? { |value| entirely_literal?(value) }
+            all_keys_entirely_literal?(node) && all_values_entirely_literal?(node)
+          when :send
+            node.method?(:freeze) && node.receiver && entirely_literal?(node.receiver)
```

Comment:
[itself](https://docs.ruby-lang.org/en/3.3/Object.html#method-i-itself) looks like a good conditate to add as well. There are others I would reasonably include as well, like `hash`, but they might have side-effects

Reply:
Actually, almost any instance method applied to a literal yields a "throwaway" result in this context (`true.to_s`, `123.to_f`, `[1,2,3].sum`, etc.)

This could be a whole new PR 👀